### PR TITLE
Update StorageClassDiagram and add use cases to DG

### LINKED ISSUE #266: [PE-D][Tester C] Billing date is restricted to only 28 days
Billing date is restricted to only 28 days, whereas by convention, each month is 30 or 31 days, and there are companied that choose to send invoices at the last day of the month

-------------
Labels: `severity.Low` `type.FeatureFlaw`
original: jinhanjinhan/ped#1

### LINKED ISSUE #303: [PE-D][Tester C] UG Bug: Hard to use prefixes as they are not documented well
Prefixes are very confusing to use. For instance, under add company command,  It says BILLING_DATE, what does that mean? Billing date refers to the date that the customers make payment to the company for their services? Or the date that the company makes salary payment?

And also, what does r/ROLE mean? Does it refer to the job that the contact currently have? Or the role that contact wants to get?

-------------
Labels: `severity.Medium` `type.DocumentationBug`
original: jinhanjinhan/ped#15

### LINKED ISSUE #340: [PE-D][Tester B] Incorrect description of test case in DG for delete contact
Command to reproduce: delete contact 1.

DG states that "timestamp in the status bar is updated":

But no such thing can be seen in the application:

-------------
Labels: `severity.VeryLow` `type.FunctionalityBug`
original: gohsl99/ped#13

### LINKED ISSUE #353: [PE-D][Tester E] Please replace this with actual acknowledgments, or just AB3
-------------
Labels: `severity.VeryLow` `type.DocumentationBug`
original: starchypotatocode/ped#30

### LINKED ISSUE #354: [PE-D][Tester E] Why is your DG file links linking to AB3 and not your own stuff?
As title

-------------
Labels: `severity.Medium` `type.DocumentationBug`
original: starchypotatocode/ped#31

### LINKED ISSUE #362: [PE-D][Tester B] Storage UML diagram is not updated in DG
As titled, in page 11 of DG:

There should also have e.g. JsonAdatpedJob, JsonAdaptedCompany etc.

-------------
Labels: `severity.VeryLow` `type.DocumentationBug`
original: gohsl99/ped#20

### PR DESCRIPTION
**Pull request by Green-Tea-123**
- [x] closes #362 
- [x] closes #354 
- [x] closes #353 
- [x] closes #340 
- [x] closes #266 
- [x] closes #303

### CODE REVIEW

File: docs/UserGuide.md [hunk 1]
```diff
@@ -111,6 +114,12 @@ Adds a company to the address book.
 
 Format: `add company n/NAME a/ADDRESS b/BILLING_DATE p/PHONE`
 
+* The name of the company is case-insensitive when used to identify duplicates 
+and to associate jobs to companies.
+* The billing date is a number from 1 to 28 that signifies the day of the month
+(with reference to the start of the month) that a company will pay their
+employees' salaries.
+
```
Comment (KengHian) [inline docs/UserGuide.md:122]: I think you can consider adding a note here mentioning end of the month option as a future enhancement, just so it is clear. 

We also need to update the Future enhancement on this as well.

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -418,7 +479,7 @@ and the **Actor** is the `user`, unless specified otherwise)
     
         Use case resumes at step 2.
 
-* 4a. The list is empty.
+* 3a. Either one or both lists are empty.
```
Comment (KengHian) [inline docs/DeveloperGuide.md:482]: I am not too sure about this, whether the lists are empty or not, the use cases ends there right? So is there a point in having this extension?

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -323,14 +323,60 @@ and the **Actor** is the `user`, unless specified otherwise)
...
+       Use case resumes at step 2
+
+**Use case: Delete company**
+
+**MSS**
+
+1. User requests to list companies
+2. TalentConnect shows a list of companies
+3. User requests to delete a specific company in the list
+4. TalentConnect deletes the company and deletes all jobs associated with the company
```
Comment (KengHian) [inline docs/DeveloperGuide.md:356]: It is probably good to explicitly mention that contacts will be "unmatched".

**Review by KengHian — APPROVED** [review]
Nice, LGTM!

File: docs/DeveloperGuide.md (hunk 2 above)
Comment (Green-Tea-123) [reply docs/DeveloperGuide.md:482]: Honestly I'm also not sure but the contacts one has it

File: docs/DeveloperGuide.md (hunk 3 above)
Comment (Green-Tea-123) [reply docs/DeveloperGuide.md:356]: ok

**Pull request merged**